tests/devices/test_default_qubit_legacy.py

```diff
@@ -652,26 +654,26 @@ def test_apply_errors_qubit_state_vector(self, qubit_device_2_wires):
             )
 
     def test_apply_errors_basis_state(self, qubit_device_2_wires):
-
-        with pytest.raises(
-            ValueError, match=r"Basis state must only consist of 0s and 1s; got \[-0\.2, 4\.2\]"
-        ):
-            qubit_device_2_wires.apply([qml.BasisState(np.array([-0.2, 4.2]), wires=[0, 1])])
-
-        with pytest.raises(
-            ValueError, match=r"State must be of length 1; got length 2 \(state=\[0 1\]\)\."
-        ):
-            qubit_device_2_wires.apply([qml.BasisState(np.array([0, 1]), wires=[0])])
-
-        with pytest.raises(
-            qml.DeviceError,
-            match="Operation BasisState cannot be used after other Operations have already been applied "
-            "on a default.qubit.legacy device.",
-        ):
-            qubit_device_2_wires.reset()
-            qubit_device_2_wires.apply(
-                [qml.RZ(0.5, wires=[0]), qml.BasisState(np.array([1, 1]), wires=[0, 1])]
-            )
+        with np.printoptions(legacy="1.21"):
```

Comment:
I wonder if it would be better to wrap these in a contextmanager + fixture? Though, fine by me to use the given style

Reply:
That's a cool idea. I think that for the limited number of tests that are impacted by this, it's not necessarily needed, but it's probably a great way to future proof the test suite. I convinced myself that I like this change as I was typing out this comment 😄

Reply:
> it's probably a great way to future proof the test suite

I think it might be something to add to the [new epic](https://app.shortcut.com/xanaduai/epic/72982?group_by=none&ct_workflow=all&cf_workflow=500000005) for CI improvements or the [epic](https://app.shortcut.com/xanaduai/epic/66006?group_by=none&vc_group_by=day&ct_workflow=all&cf_workflow=500000005) for improving robustness of tests + CI. I'll resolve the conversation since It seems out of the scope of the original PR.

Reply:
The 2nd epic you linked is probably more appropriate. The first one is more targeted towards updating the workflows, not the tests.